### ISSUE TITLE
[Question] Can I send commands to an existing Alacritty instance (not during start-up)?

### ISSUE THREAD

**Issue by jasonhan-vassar** [alacritty/alacritty#5784]
Hi, I'd like to send commands into an Alacritty instance from another program. Is this supported?

My use case is to use Alacritty as a "build system" for my text editor — I want to reuse the same terminal window to avoid the delay of starting up a new one with the `-e` flag.

**Comment by chrisduerr** [comment]
Not really, no. Not something we'd like to support either.

**Comment by jasonhan-vassar** [comment]
Fair enough. After asking this, I did discover that kitty has this feature and tried it a bit, but it seems rather slow and complex. Is this the main reason not to add it to Alacritty?

**Comment by chrisduerr** [comment]
> After asking this, I did discover that kitty has this feature and tried it a bit, but it seems rather slow and complex.

While Kitty might be more complex than Alacritty, I'm not aware of it being particularly slow, at least compared to most other non-opengl terminals.

> After asking this, I did discover that kitty has this feature and tried it a bit, but it seems rather slow and complex.

No, this has no performance impact. It's just not a good idea to allow arbitrary other applications to mess with user input.

**Comment by lieskjur** [comment]
@chrisduerr I would suggest that your concern might actually be a good reason to implement such a feature. I know that Wayland has strict security policies when it comes to input, but Xorg does not. Which means that through the use of xdotools you can do pretty much whatever you want.

I for example use a plugin that sends individual lines or entire blocks of code from my text editor into a REPL running in alacritty. This has greatly increased my productivity when writing scripts in python and julia as it is bound to a single keyboard shortcut and uses regex to expand the selection. I am also not the only one using such tools as both vscode and matlab's IDE have similar features.

In conclusion, while causing some necessary evil, such a feature would deter hackier solutions while salvaging some tools in the future. But I would like to know your opinion on the feasibility of making such a feature reasonably safe.
